packages/edit-site/src/components/sidebar/typography-panel.js

```diff
@@ -61,7 +74,18 @@ export default function TypographyPanel( {
 		'typography.customFontSize',
 		name
 	);
-	const fontFamilies = useSetting( 'typography.fontFamilies', name );
+	const coreFontFamilies = useSetting( 'typography.fontFamilies.core', name );
+	const userFontFamilies = useSetting( 'typography.fontFamilies.user', name );
+	const themeFontFamilies = useSetting(
+		'typography.fontFamilies.theme',
+		name
+	);
+	const fontFamilies = mergeWithoutDuplicateFontFamilies(
+		themeFontFamilies,
+		userFontFamilies,
+		coreFontFamilies
+	);
```

Comment:
Nit: I think we can use `_.unionBy(themeFontFamilies, userFontFamilies, coreFontFamilies, 'slug');` here.